stan/math/prim/scal/fun/as_bool.hpp

```diff
@@ -5,13 +5,14 @@ namespace stan {
   namespace math {
 
     /**
-     * Return 1 if the argument is unequal to zero and 0 otherwise.
+     * Return true if the argument is unequal to zero and false otherwise.
      *
-     * @param x Value.
-     * @return 1 if argument is equal to zero (or NaN) and 0 otherwise.
+     * @tparam type of scalar
+     * @param x value
+     * @return true if value is equal to zero or NaN
```

Comment:
Yes, that's backwards.  I'll fix it.   

Depends what you mean by equal.  If you mean `operator==`, then that's right, `NaN == NaN` returns false, which is why we have `is_nan()`.  I'll try to rephrase to make sure the "equal to" isn't confused as being `==` applied to NaN.